### ISSUE TITLE
Merge Storage-NIO library into this repository.

### ISSUE THREAD

**Issue by frankyn** [googleapis/java-storage#242]
Currently have two repositories that must  be merged into one library.

https://github.com/googleapis/java-storage-nio needs to be merged into this repository to unblock changes to the underlying internal only StorageRpc interface. Otherwise we can accidentally break users.

**Comment by frankyn** [comment]
After discussing with @chingor13, the more likely approach is to separate out testing related code used by the Storage-Nio library into a separate artifact called google-cloud-storage-testing instead of merging Storage-NIO library into this repository.

**Comment by dmitry-fa** [comment]
@frankyn, I this is about exporting FakeStorageRpc for the sake of testing I can take care of this issue to move it on. #202 and #179 are currently blocked by this issue.

**Comment by dmitry-fa** [comment]
Suggested fix #345 creates a copy of `FakeStorageRpc` from Storage-Nio in Storage. Because the main storage artifact already includes a testing package (`com.google.cloud.storage.testing`) the new class is placed there. No extra artifact is required. 

I verified, Storage-Nio could switch to the new `FakeStorageRpc` class without adding extra dependencies.

**Comment by dmitry-fa** [comment]
`FakeStorageRpc` from #345 is very NIO specific, unlikely it will be used by anyone else. #351 provides a more generic approach: `StorageRpcTestBase` class implementing `StorageRpc`. `FakeStorageRpc` can extend `StorageRpcTestBase`. 

cc: @frankyn
